src/components/Settings.js

```diff
@@ -20,16 +26,19 @@ class Settings extends React.Component {
                 this.props.submitUserName(input.value)
                 mainProcess.saveUsername({ username: input.value })
                 }}>
-            <section className='mod-input'>
-              <label>
-                <h2>Github Username</h2>
-              </label>
-              <input placeholder='Github username' ref={ node => {input = node} } />
-              <button children='Submit' />
+            <section aria-labelledby='input-label'>
+              <h2 className='input-label'>Please enter your Github username to Proceed:</h2>
+              <input
+                placeholder='Github username'
+                ref={ node => {input = node} }
+                onKeyUp={(e)=> this.setState({fieldValue: e.target.value})
+              }/>
+              <button children='Submit' disabled={!this.state.fieldValue} />
             </section>
           </form>
           </section>
-        <Navigation forward={true} route={'/home'}/>
+          { this.props.username ?
+              <Navigation forward={true} route={'/home'}/> : null }
       </section>
```

Comment:
Interesting approach. I have removed the Navigation entirely in an iteration that I am currently working on. Do you care which way we go with this when we are done?